source/val/validate_memory.cpp

```diff
@@ -502,6 +502,82 @@ spv_result_t ValidateVariable(ValidationState_t& _, const Instruction* inst) {
     }
   }
 
+  // Vulkan specific validation rules
+  if (spvIsVulkanEnv(_.context()->target_env)) {
+    const auto type_index = 2;
+    const auto value_id = result_type->GetOperandAs<uint32_t>(type_index);
+    auto value_type = _.FindDef(value_id);
+    // OpTypeRuntimeArray should only ever be in an OpTypeStruct, so should
+    // never appear as a bare variable.
```

Comment:
This looks like it doesn't handle the changes from SPV_EXT_descriptor_indexing, which also allows runtime arrays of descriptor variables. The Vulkan rules currently say:

```
  * code:OpTypeRuntimeArray must: only be used for:
  ** the last member of an code:OpTypeStruct
ifdef::VK_VERSION_1_1,VK_KHR_storage_buffer_storage_class[]
     that is in the code:StorageBuffer storage class decorated as
     code:Block, or
endif::VK_VERSION_1_1,VK_KHR_storage_buffer_storage_class[]
     that is in the code:Uniform storage class decorated as
     code:BufferBlock.
ifdef::VK_EXT_descriptor_indexing[]
  ** If the code:RuntimeDescriptorArrayEXT capability is supported, an array
     of variables with storage class code:Uniform,
ifdef::VK_VERSION_1_1,VK_KHR_storage_buffer_storage_class[]
     code:StorageBuffer,
endif::VK_VERSION_1_1,VK_KHR_storage_buffer_storage_class[]
     or code:UniformConstant, or for the outermost dimension of an array of
     arrays of such variables.
endif::VK_EXT_descriptor_indexing[]
```

Reply:
Thanks for pointing this out. I have updated the PR to handle this case.